### PR TITLE
Repackaging 2.0

### PR DESCRIPTION
**Pull request by mhevery**
- [x] rebase
- [x] update tscikle to be compatible with tsc@next
- [x] update tslint to the `@dev` tag to be compatible with tsc@next
- [x] shrinkwrap typescript
- [x] ? bundle all testing stuff into a single file? (outDir -> outFile)
- [x]  resolve any CI issues
- [x] double check all package.json's `module` and `typings` entries
- [ ] verify rollup's `globals` configuration for RxJS

### CODE REVIEW

**Comment by mhevery** [comment]
I attempted to shrinkwrap by cloning https://github.com/mhevery/tsickle and https://github.com/mhevery/tslint, but it did not seem to work. Will wait on monday.

File: tools/public_api_guard/router/index.d.ts [hunk 1]
```diff
@@ -67,9 +67,6 @@ export declare class DefaultUrlSerializer implements UrlSerializer {
 }
 
 /** @stable */
-export declare type ErrorHandler = (error: any) => any;
```
Comment (IgorMinar) [inline tools/public_api_guard/router/index.d.ts]: is this intentional?

File: tools/@angular/tsc-wrapped/src/main.ts [hunk 2]
```diff
@@ -28,6 +28,11 @@ export function main(
     ngOptions.basePath = basePath;
 
     const host = ts.createCompilerHost(parsed.options, true);
+
+    // HACK: patch the realpath to solve symlink issue here:
+    // https://github.com/Microsoft/TypeScript/issues/9552
```
Comment (IgorMinar) [inline tools/@angular/tsc-wrapped/src/main.ts]: can you add a note that we can remove this once we get rid of symlinked facades

File: test-main.js [hunk 3]
```diff
@@ -13,21 +13,34 @@ System.config({
   map: {
     'benchpress/*': 'dist/js/dev/es5/benchpress/*.js',
     '@angular': 'dist/all/@angular',
+    '@angular/core': 'dist/all/@angular/core',
```
Comment (IgorMinar) [inline test-main.js]: I don't think you want this line. remove

File: scripts/ci-lite/offline_compiler_test.sh [hunk 4]
```diff
@@ -13,6 +13,7 @@ PKGS=(
   rxjs
   @types/{node,jasmine}
   jasmine
+  webpack@beta
```
Comment (IgorMinar) [inline scripts/ci-lite/offline_compiler_test.sh]: can you pin it to a particular version so that it doesn't fail in the future without us knowing

File: scripts/ci-lite/build.sh [hunk 5]
```diff
@@ -13,7 +13,22 @@ $(npm bin)/tsc -v
...
+  platform-browser-dynamic \
+  platform-server \
+  http \
+  router \
+  upgrade \
+  compiler-cli
+do
+  node dist/tools/@angular/tsc-wrapped/src/main -p modules/@angular/$PACKAGE
+  cp  modules/@angular/$PACKAGE/package.json dist/packages-dist/$PACKAGE
+done
```
Comment (IgorMinar) [inline scripts/ci-lite/build.sh]: why this change? the top level build.sh does this already.

File: modules/@angular/upgrade/rollup.config.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 
 export default {
-  entry: '../../../dist/packages-dist/upgrade/esm/index.js',
-  dest: '../../../dist/packages-dist/upgrade/esm/upgrade.umd.js',
+  entry: '../../../dist/packages-dist/upgrade/index.js',
+  dest: '../../../dist/packages-dist/upgrade/upgrade.umd.js',
```
Comment (IgorMinar) [inline modules/@angular/upgrade/rollup.config.js]: update to `bundles/upgrade.umd.js`

File: modules/@angular/router/tsconfig.json [hunk 7]
```diff
@@ -3,39 +3,26 @@
     "baseUrl": ".",
     "declaration": true,
     "stripInternal": true,
-    "emitDecoratorMetadata": true,
```
Comment (IgorMinar) [inline modules/@angular/router/tsconfig.json]: keep

File: modules/@angular/router/tsconfig.json [hunk 8]
```diff
@@ -3,39 +3,26 @@
...
     "experimentalDecorators": true,
-    "module": "commonjs",
+    "module": "es2015",
     "moduleResolution": "node",
-    "noEmitOnError": false,
-    "noImplicitAny": true,
-    "noImplicitReturns": true,
```
Comment (IgorMinar) [inline modules/@angular/router/tsconfig.json]: keep both  noImplicitAny and noImplicitReturns

Comment (IgorMinar) [inline modules/@angular/router/tsconfig.json]: add these to the router/tsconfig-testing.json as well

File: modules/@angular/router/testing/router_testing_module.ts [hunk 9]
```diff
@@ -9,15 +9,19 @@
...
-import {Route, Router, RouterOutletMap, UrlSerializer} from '../index';
-import {Routes} from '../src/config';
-import {ROUTES} from '../src/router_config_loader';
-import {ROUTER_PROVIDERS, RouterModule, provideRoutes} from '../src/router_module';
-import {flatten} from '../src/utils/collection';
-
-
+import {ROUTER_PROVIDERS, ROUTES} from './router_private';
 
+export function flatten<T>(a: T[][]): T[] {
```
Comment (IgorMinar) [inline modules/@angular/router/testing/router_testing_module.ts]: since you have router_private, you can export the existing code from there rather than duplicate it in this way.

File: modules/@angular/router/ng_probe_token.ts [hunk 10]
```diff
@@ -1,16 +0,0 @@
...
- */
-
-import {NgProbeToken} from '@angular/platform-browser';
-import {Router} from './src/router';
-
-export const ROUTER_NG_PROBE_PROVIDER = {
-  provide: NgProbeToken,
-  multi: true,
-  useValue: new NgProbeToken('router', Router)
-};
```
Comment (IgorMinar) [inline modules/@angular/router/ng_probe_token.ts]: what happened to this provider? it seems to have gone missing.

File: modules/@angular/platform-server/tsconfig.json [hunk 11]
```diff
@@ -4,24 +4,26 @@
...
+    "outDir": "../../../dist/packages-dist/platform-server",
     "paths": {
-      "@angular/core": ["../../../dist/packages-dist/core"],
-      "@angular/common": ["../../../dist/packages-dist/common"],
-      "@angular/compiler": ["../../../dist/packages-dist/compiler"],
-      "@angular/platform-browser": ["../../../dist/packages-dist/platform-browser"],
+      "@angular/core": ["../../../dist/packages-dist/core/"],
+      "@angular/common": ["../../../dist/packages-dist/common/"],
+      "@angular/compiler": ["../../../dist/packages-dist/compiler/"],
+      "@angular/platform-browser": ["../../../dist/packages-dist/platform-browser/"],
```
Comment (IgorMinar) [inline modules/@angular/platform-server/tsconfig.json]: remove trailing slashes in the 4 lines above to be consistent

File: modules/@angular/platform-server/src/server.ts [hunk 12]
```diff
@@ -32,6 +32,9 @@ class ServerPlatformLocation extends PlatformLocation {
   back(): void { notSupported('back'); };
 }
 
+/**
+ * @experimental
+ */
```
Comment (IgorMinar) [inline modules/@angular/platform-server/src/server.ts]: this is not public api. remove the `@experimental` tag

File: modules/@angular/platform-server/src/parse5_adapter.ts [hunk 13]
```diff
@@ -9,9 +9,9 @@
 var parse5 = require('parse5/index');
 
 import {ListWrapper, StringMapWrapper} from '../src/facade/collection';
-import {DomAdapter, setRootDomAdapter} from '../platform_browser_private';
+import {DomAdapter, setRootDomAdapter} from '../testing/platform_browser_private';
```
Comment (IgorMinar) [inline modules/@angular/platform-server/src/parse5_adapter.ts]: this is not right. the `src` code should not import from `testing/`

File: modules/@angular/platform-server/rollup-testing.config.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+  globals: {
+    '@angular/core': 'ng.core',
+    '@angular/common': 'ng.common',
+    '@angular/compiler': 'ng.compiler',
+    '@angular/compiler/testing': 'ng.compiler.testing',
+    '@angular/platform-browser': 'ng.platformBrowser',
+    '@angular/platform-server': 'ng.platformServer',
+    '@angular/platform-browser-dynamic/testing': 'ng.platformBrowserDynamic.testing',
+    'rxjs/Subject': 'Rx',
+    'rxjs/observable/PromiseObservable': 'R x', // this is wrong, but this stuff has changed in rxjs b.6 so we need to fix it when we update.
```
Comment (IgorMinar) [inline modules/@angular/platform-server/rollup-testing.config.js]: "R x"? at least we know that this is totally broken

File: modules/@angular/platform-browser/testing/browser_util.ts [hunk 15]
```diff
@@ -7,9 +7,11 @@
...
 
 import {NgZone} from '@angular/core';
-import {getDOM} from '../src/dom/dom_adapter';
-import {ListWrapper} from '../src/facade/collection';
-import {RegExp, StringWrapper, global, isPresent, isString} from '../src/facade/lang';
+import {__platform_browser_private__} from '@angular/platform-browser';
+import {ListWrapper} from './facade/collection';
+import {RegExp, StringWrapper, global, isPresent, isString} from './facade/lang';
+
+const {getDOM} = __platform_browser_private__;
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/testing/browser_util.ts]: don't we have local file to import this from? we should be consistent. can you redo?

File: modules/@angular/platform-browser/testing/core_private_testing.ts [hunk 16]
```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {__core_private__ as r} from '@angular/core';
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/testing/core_private_testing.ts]: since this file imports from core, it should be called core_private.ts. maybe a better name convention would be `private_import_<src>.ts` to map to `private_export.ts`. so in this case it would be `private_import_core.ts`

File: modules/@angular/platform-browser/src/dom/debug/ng_probe.ts [hunk 17]
```diff
@@ -60,6 +60,8 @@ function _ngProbeTokensToMap(tokens: NgProbeToken[]): {[name: string]: any} {
 
 /**
  * Providers which support debugging Angular applications (e.g. via `ng.probe`).
+ *
+ * @experimental
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/dom/debug/ng_probe.ts]: please remove - this is not a public symbol

File: modules/@angular/platform-browser/src/browser/browser_adapter.ts [hunk 18]
```diff
@@ -67,6 +67,8 @@ var _chromeNumKeyPadMap = {
  *
  * @security Tread carefully! Interacting with the DOM directly is dangerous and
  * can introduce XSS risks.
+ *
+ * @experimental
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/src/browser/browser_adapter.ts]: please remove - not a public api

File: modules/@angular/platform-browser/package.json [hunk 19]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@angular/platform-browser",
   "version": "0.0.0-PLACEHOLDER",
-  "description": "",
-  "main": "index.js",
-  "jsnext:main": "esm/index.js",
+  "description": "Angular 2 platfor-browser",
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/package.json]: typo. platform-browser

File: modules/@angular/platform-browser-dynamic/testing/platform_browser_testing_private.ts [hunk 20]
```diff
@@ -7,7 +7,11 @@
  */
 
 import {__platform_browser_private__ as r} from '@angular/platform-browser';
+import {__platform_browser_dynamic_private__ as dr} from '@angular/platform-browser-dynamic';
```
Comment (IgorMinar) [inline modules/@angular/platform-browser-dynamic/testing/platform_browser_testing_private.ts]: this should be split into two files to be consistent

File: modules/@angular/core/testing/fake_async.ts [hunk 21]
```diff
@@ -87,6 +86,7 @@ export function fakeAsync(fn: Function): (...args: any[]) => any {
       return res;
     } finally {
       _inFakeAsyncCall = false;
+      resetFakeAsyncZone();
```
Comment (IgorMinar) [inline modules/@angular/core/testing/fake_async.ts]: what is this change doing here?

File: modules/@angular/core/src/application_ref.ts [hunk 22]
```diff
@@ -283,11 +276,11 @@ export class PlatformRef_ extends PlatformRef {
       const moduleRef = <NgModuleInjector<M>>moduleFactory.create(ngZoneInjector);
       const exceptionHandler: ErrorHandler = moduleRef.injector.get(ErrorHandler, null);
       if (!exceptionHandler) {
-        throw new Error('No ExceptionHandler. Is platform module (BrowserModule) included?');
+        throw new Error('No ErrorHandler. Is platform module (BrowserModule) included?');
```
Comment (IgorMinar) [inline modules/@angular/core/src/application_ref.ts]: did you just see these leftover instances of ExceptionHandler usage and fix them or how did these changes get here?

File: modules/@angular/compiler/rollup-testing.config.js [hunk 23]
```diff
@@ -0,0 +1,19 @@
...
+  globals: {
+    '@angular/core': 'ng.core',
+    '@angular/core/testing': 'ng.core.testing',
+    '@angular/compiler': 'ng.compiler',
+    'rxjs/Subject': 'Rx',
+    'rxjs/observable/PromiseObservable': 'Rx', // this is wrong, but this stuff has changed in rxjs b.6 so we need to fix it when we update.
+    'rxjs/operator/toPromise': 'Rx.Observable.prototype',
+    'rxjs/Observable': 'Rx'
+  },
+  plugins: [
```
Comment (IgorMinar) [inline modules/@angular/compiler/rollup-testing.config.js]: remove the empty plugins section

File: modules/@angular/compiler/package.json [hunk 24]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "@angular/compiler",
   "version": "0.0.0-PLACEHOLDER",
-  "description": "",
-  "main": "index.js",
-  "jsnext:main": "esm/index.js",
+  "description": "Angular2 - core",
```
Comment (IgorMinar) [inline modules/@angular/compiler/package.json]: core - > compiler

File: modules/@angular/compiler-cli/test/reflector_host_spec.ts [hunk 25]
```diff
@@ -13,7 +13,7 @@ import {ReflectorHost} from '../src/reflector_host';
 
 import {Directory, Entry, MockCompilerHost, MockContext} from './mocks';
 
-describe('reflector_host', () => {
+ddescribe('reflector_host', () => {
```
Comment (IgorMinar) [inline modules/@angular/compiler-cli/test/reflector_host_spec.ts]: ddescribe

File: build.sh [hunk 26]
```diff
@@ -74,50 +74,49 @@ do
...
-        --lib "es6,dom" \
-        --allowJs \
-        ${UMD_ES6_PATH}
-
-    rm ${UMD_ES6_PATH}
 
-    cat ./modules/@angular/license-banner.txt > ${UMD_ES5_PATH}.tmp
-    cat ${UMD_ES5_PATH} >> ${UMD_ES5_PATH}.tmp
-    mv ${UMD_ES5_PATH}.tmp ${UMD_ES5_PATH}
+    # cat ./modules/@angular/license-banner.txt > ${UMD_ES5_PATH}.tmp
```
Comment (IgorMinar) [inline build.sh]: you still need to uncomment this and configure bundle minificatio

File: build.sh [hunk 27]
```diff
@@ -74,50 +74,49 @@ do
...
-    echo "====== (esm)COMPILING: $TSC -p ${SRCDIR}/tsconfig-es2015.json ====="
-    $TSC -p ${SRCDIR}/tsconfig-es2015.json
-
     echo "======      BUNDLING: ${SRCDIR} ====="
     mkdir ${DESTDIR}/bundles
 
     (
       cd  ${SRCDIR}
-      echo "..."  # here just to have grep match something and not exit with 1
+      echo "======         Rolup ${PACKAGE} index"
```
Comment (IgorMinar) [inline build.sh]: Rolup -> Rollup

File: build.sh [hunk 28]
```diff
@@ -74,50 +74,49 @@ do
...
       ../../../node_modules/.bin/rollup -c rollup.config.js
+      if [[ -e rollup-testing.config.js ]]; then
+        echo "======         Rolup ${PACKAGE} testing"
```
Comment (IgorMinar) [inline build.sh]: Rolup -> Rollup

File: tools/public_api_guard/router/index.d.ts (hunk 1 above)
Comment (mhevery) [inline tools/public_api_guard/router/index.d.ts]: yes, ErrorHandler is exported by `core` not by `router`.

File: tools/@angular/tsc-wrapped/src/main.ts (hunk 2 above)
Comment (mhevery) [inline tools/@angular/tsc-wrapped/src/main.ts]: done

File: test-main.js (hunk 3 above)
Comment (mhevery) [inline test-main.js]: remvode

File: scripts/ci-lite/offline_compiler_test.sh (hunk 4 above)
Comment (mhevery) [inline scripts/ci-lite/offline_compiler_test.sh]: set to `webpack@2.1.0-beta.21`

File: modules/@angular/upgrade/rollup.config.js (hunk 6 above)
Comment (mhevery) [inline modules/@angular/upgrade/rollup.config.js]: will do in a separate commit.

File: modules/@angular/router/tsconfig.json (hunk 7 above)
Comment (mhevery) [inline modules/@angular/router/tsconfig.json]: done

File: modules/@angular/router/tsconfig.json (hunk 8 above)
Comment (mhevery) [inline modules/@angular/router/tsconfig.json]: done

File: modules/@angular/router/testing/router_testing_module.ts (hunk 9 above)
Comment (mhevery) [inline modules/@angular/router/testing/router_testing_module.ts]: done

File: modules/@angular/router/ng_probe_token.ts (hunk 10 above)
Comment (mhevery) [inline modules/@angular/router/ng_probe_token.ts]: As far as I could tell it was not part of public API and no one was referring to it.

File: modules/@angular/platform-server/tsconfig.json (hunk 11 above)
Comment (mhevery) [inline modules/@angular/platform-server/tsconfig.json]: done

File: modules/@angular/platform-server/src/parse5_adapter.ts (hunk 13 above)
Comment (mhevery) [inline modules/@angular/platform-server/src/parse5_adapter.ts]: IDE, messed up, fixed

File: modules/@angular/platform-server/rollup-testing.config.js (hunk 14 above)
Comment (mhevery) [inline modules/@angular/platform-server/rollup-testing.config.js]: fixed

File: modules/@angular/platform-browser/testing/browser_util.ts (hunk 15 above)
Comment (mhevery) [inline modules/@angular/platform-browser/testing/browser_util.ts]: I don't understand what you are asking me to do. Be more specific.

File: modules/@angular/platform-browser/testing/core_private_testing.ts (hunk 16 above)
Comment (mhevery) [inline modules/@angular/platform-browser/testing/core_private_testing.ts]: done

File: modules/@angular/platform-browser/src/dom/debug/ng_probe.ts (hunk 17 above)
Comment (mhevery) [inline modules/@angular/platform-browser/src/dom/debug/ng_probe.ts]: fixed

File: modules/@angular/platform-browser/src/browser/browser_adapter.ts (hunk 18 above)
Comment (mhevery) [inline modules/@angular/platform-browser/src/browser/browser_adapter.ts]: fixed

File: modules/@angular/platform-browser/package.json (hunk 19 above)
Comment (mhevery) [inline modules/@angular/platform-browser/package.json]: fixed

File: modules/@angular/platform-browser-dynamic/testing/platform_browser_testing_private.ts (hunk 20 above)
Comment (mhevery) [inline modules/@angular/platform-browser-dynamic/testing/platform_browser_testing_private.ts]: fixed

File: modules/@angular/core/testing/fake_async.ts (hunk 21 above)
Comment (mhevery) [inline modules/@angular/core/testing/fake_async.ts]: tests were failing without it. Not sure why it got triggered now, but that method call was not invoked from anywhere.

File: modules/@angular/core/src/application_ref.ts (hunk 22 above)
Comment (mhevery) [inline modules/@angular/core/src/application_ref.ts]: Yes, we missed them in previous pass.

File: modules/@angular/compiler/rollup-testing.config.js (hunk 23 above)
Comment (mhevery) [inline modules/@angular/compiler/rollup-testing.config.js]: done

File: modules/@angular/compiler/package.json (hunk 24 above)
Comment (mhevery) [inline modules/@angular/compiler/package.json]: fixed

File: modules/@angular/compiler-cli/test/reflector_host_spec.ts (hunk 25 above)
Comment (mhevery) [inline modules/@angular/compiler-cli/test/reflector_host_spec.ts]: fixed

File: build.sh (hunk 26 above)
Comment (mhevery) [inline build.sh]: will do in next SHA

File: build.sh (hunk 27 above)
Comment (mhevery) [inline build.sh]: done

File: build.sh (hunk 28 above)
Comment (mhevery) [inline build.sh]: done

File: modules/@angular/router/ng_probe_token.ts (hunk 10 above)
Comment (IgorMinar) [inline modules/@angular/router/ng_probe_token.ts]: @vsavkin is this intentional?

File: modules/@angular/platform-browser/testing/browser_util.ts (hunk 15 above)
Comment (IgorMinar) [inline modules/@angular/platform-browser/testing/browser_util.ts]: you are importing directly from `@angular/platform-browser` instead of platform_browser_private. it looks like the import side of the private channel was never setup. can you do it?

File: build.sh [hunk 29]
```diff
@@ -105,18 +108,25 @@ do
       cd  ${SRCDIR}
       echo "======         Rollup ${PACKAGE} index"
       ../../../node_modules/.bin/rollup -c rollup.config.js
+      cat ${LICENSE_BANNER} > ${UMD_ES5_PATH}.tmp
+      cat ${UMD_ES5_PATH} >> ${UMD_ES5_PATH}.tmp
+      mv ${UMD_ES5_PATH}.tmp ${UMD_ES5_PATH}
+      $UGLIFYJS -c --screw-ie8 -o ${UMD_ES5_MIN_PATH} ${UMD_ES5_PATH}
```
Comment (IgorMinar) [inline build.sh]: please add  `--comments` to preserve the licensing comment

File: modules/@angular/platform-browser-dynamic/src/platform_browser_private.ts [hunk 30]
```diff
@@ -6,8 +6,8 @@
  * found in the LICENSE file at https://angular.io/license
```
Comment (IgorMinar) [inline modules/@angular/platform-browser-dynamic/src/platform_browser_private.ts]: rename this file to private_import_platform-browser

File: modules/@angular/platform-browser/testing/platform_browser_private.ts [hunk 31]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/testing/platform_browser_private.ts]: rename to private_import_platform-browser

File: modules/@angular/router/testing/router_private.ts [hunk 32]
```diff
@@ -0,0 +1,13 @@
+/**
```
Comment (IgorMinar) [inline modules/@angular/router/testing/router_private.ts]: rename to private_import_router.ts

File: modules/@angular/compiler-cli/src/compiler_private.ts [hunk 33]
```diff
@@ -6,50 +6,49 @@
  * found in the LICENSE file at https://angular.io/license
  */
```
Comment (IgorMinar) [inline modules/@angular/compiler-cli/src/compiler_private.ts]: rename to private_import_compiler.ts

File: modules/@angular/platform-browser/testing/browser_util.ts (hunk 15 above)
Comment (mhevery) [inline modules/@angular/platform-browser/testing/browser_util.ts]: done

File: modules/@angular/platform-browser-dynamic/src/platform_browser_private.ts (hunk 30 above)
Comment (mhevery) [inline modules/@angular/platform-browser-dynamic/src/platform_browser_private.ts]: done

File: modules/@angular/platform-browser/testing/platform_browser_private.ts (hunk 31 above)
Comment (mhevery) [inline modules/@angular/platform-browser/testing/platform_browser_private.ts]: done

File: modules/@angular/router/testing/router_private.ts (hunk 32 above)
Comment (mhevery) [inline modules/@angular/router/testing/router_private.ts]: done

File: modules/@angular/compiler-cli/src/compiler_private.ts (hunk 33 above)
Comment (mhevery) [inline modules/@angular/compiler-cli/src/compiler_private.ts]: done

File: build.sh (hunk 29 above)
Comment (mhevery) [inline build.sh]: done

File: scripts/ci-lite/build.sh (hunk 5 above)
Comment (IgorMinar) [inline scripts/ci-lite/build.sh]: was this addressed?

File: scripts/ci-lite/build.sh [hunk 34]
```diff
@@ -12,8 +12,7 @@ cd ../..
 $(npm bin)/tsc -v
 $(npm bin)/tsc -p tools
 cp tools/@angular/tsc-wrapped/package.json dist/tools/@angular/tsc-wrapped
-node dist/tools/@angular/tsc-wrapped/src/main -p modules
-node dist/tools/@angular/tsc-wrapped/src/main -p modules/@angular/router
+./build.sh
```
Comment (IgorMinar) [inline scripts/ci-lite/build.sh]: this will slow down the CI significantly. can we revert it? why is it needed?

File: tools/tsc-watch/index.ts [hunk 35]
```diff
@@ -79,24 +79,39 @@ if (platform == 'node') {
...
+          'modules/@angular/router/karma.conf.js'
+        ],
+      ],
+      onChangeCmds: [
+        ['node', 'node_modules/karma/bin/karma', 'run', 'karma-js.conf.js', '--port=9876'],
+        ['node', 'node_modules/karma/bin/karma', 'run', '--port=9877'],
+      ]
+    },
+    BaseConfig));
+} else if (platform == 'router') {
```
Comment (IgorMinar) [inline tools/tsc-watch/index.ts]: what happened here?

File: package.json [hunk 36]
```diff
@@ -109,9 +109,9 @@
     "through2": "^0.6.5",
     "ts-api-guardian": "0.1.4",
     "ts2dart": "^0.9.10",
-    "tsickle": "^0.1.4",
-    "tslint": "^3.10.0-dev.2",
-    "typescript": "^1.9.0-dev.20160409",
+    "tsickle": "^0.1.7",
+    "tslint": "^3.15.0-dev.0",
+    "typescript": "^2.1.0-dev.20160827",
```
Comment (IgorMinar) [inline package.json]: let's update to rc

File: modules/@angular/upgrade/package.json [hunk 37]
```diff
@@ -2,8 +2,9 @@
   "name": "@angular/upgrade",
   "version": "0.0.0-PLACEHOLDER",
   "description": "",
-  "main": "upgrade.js",
-  "jsnext:main": "esm/upgrade.js",
+  "main": "index.js",
```
Comment (IgorMinar) [inline modules/@angular/upgrade/package.json]: this is wrong. the main should point to bundle, module to index.js and typings to index.d.ts

File: modules/@angular/platform-browser/testing/matchers.ts [hunk 38]
```diff
@@ -6,10 +6,11 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {getDOM} from '../src/dom/dom_adapter';
-import {StringMapWrapper} from '../src/facade/collection';
-import {global, isString} from '../src/facade/lang';
+import {__platform_browser_private__} from '@angular/platform-browser';
```
Comment (IgorMinar) [inline modules/@angular/platform-browser/testing/matchers.ts]: import from private_import_platform-browser

File: scripts/ci-lite/build.sh (hunk 34 above)
Comment (mhevery) [inline scripts/ci-lite/build.sh]: trying revert

File: tools/tsc-watch/index.ts (hunk 35 above)
Comment (mhevery) [inline tools/tsc-watch/index.ts]: added new route comment

File: package.json (hunk 36 above)
Comment (mhevery) [inline package.json]: separate SHA

File: modules/@angular/upgrade/package.json (hunk 37 above)
Comment (mhevery) [inline modules/@angular/upgrade/package.json]: fixed

File: modules/@angular/platform-browser/testing/matchers.ts (hunk 38 above)
Comment (mhevery) [inline modules/@angular/platform-browser/testing/matchers.ts]: fixed

File: build.sh [hunk 39]
```diff
@@ -68,56 +69,61 @@ for PACKAGE in \
...
+    $TSC -p ${SRCDIR}/tsconfig-testing.json
+  fi
 
   echo "======      TSC 1.8 d.ts compat for ${DESTDIR}   ====="
   # safely strips 'readonly' specifier from d.ts files to make them compatible with tsc 1.8
   if [ "$(uname)" == "Darwin" ]; then
-    find ${DESTDIR} -type f -name '*.d.ts' -print0 | xargs -0 sed -i ''    -e 's/\(^ *(static |private )*\)*readonly  */\1/g'
-    find ${DESTDIR} -type f -name '*.d.ts' -print0 | xargs -0 sed -i ''    -E 's/^( +)abstract ([[:alnum:]]+\:)/\1\2/g'
+    find ${DESTDIR} -type f -name '*.d.ts' -print0 | xargs -0 sed -i '' -e 's/\(^ *(static |private )*\)*readonly  */\1/g'
+    find ${DESTDIR} -type f -name '*.d.ts' -print0 | xargs -0 sed -i '' -e 's/\/\/\/ <reference types="node" \/>//g'
```
Comment (alexeagle) [inline build.sh]: + @chuckjaz  :)

**Comment by Martin-Wegner** [comment]
Since nightly build https://github.com/angular/core-builds/commit/8509b8b39c67a80fa57d1ced70527896d6ca92cd (which should be this pull request I think) I got this error when testing:

> PhantomJS 2.1.1 (Linux 0.0.0) ERROR
>   SyntaxError: Use of reserved word 'export'
>   at webpack:///~/@angular/core/src/facade/lang.js:21:0 <- spec-bundle.js:72521

Any suggestions?

**Comment by Martin-Wegner** [comment]
OK my whole application does not work anymore :(

> lang.js?c27c:21 Uncaught SyntaxError: Unexpected token export

**Comment by IgorMinar** [comment]
Martin, make sure that you don't have any deep imports into the "src"
directory within angular packages. You should only import stuff from the
top level entry points (e.g. @angular/core, @angular/core/testing).

in your case it looks like you are using webpack 1, which should by default
consume the umd bundles as directed via package.json's main entry. The main
reason why this wouldn't work is when you have a deep import into the
package.

**Comment by Martin-Wegner** [comment]
@IgorMinar  thanks, you are right, the ng2-bootstrap folks use deep imports :(

**Comment by pkozlowski-opensource** [comment]
@Martin-Wegner this would be a good time to explore a library that is not using private APIs: https://github.com/ng-bootstrap/ng-bootstrap :-)

**Comment by Martin-Wegner** [comment]
@pkozlowski-opensource thanks for the hint, but your issue https://github.com/ng-bootstrap/ng-bootstrap/issues/646 scares me from using it...

**Comment by pkozlowski-opensource** [comment]
ng-bootstrap/ng-bootstrap#646 is for our demo page, nothing to do with the library itself.
